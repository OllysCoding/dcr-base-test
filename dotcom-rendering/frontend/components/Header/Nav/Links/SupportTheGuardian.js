// @flow

import palette, { pillars } from '@guardian/pasteup/palette';
import { headline } from '@guardian/pasteup/fonts';
import { cx, css } from 'react-emotion';

const style = css`
    color: ${palette.neutral.header};
    font-family: ${headline};
    font-size: 14px;
    font-weight: 500;
    text-decoration: none;
    position: relative;
    width: 135px;
    display: block;
    text-align: center;
    padding-top: 6px;
    z-index: 0;
    float: left;
    :before {
        background-color: ${palette.neutral['1']};
        border-radius: 50%;
        top: -85px;
        left: 0;
        right: 0;
        padding-top: 100%;
        content: '';
        display: block;
        position: absolute;
        transition: background-color 250ms ease-out;
        z-index: -1;
    }
    :hover:before {
        background-color: ${pillars.news};
        transform: scale(1.05);
    }
`;
const SupportTheGuardian = ({
    className,
    children,
    ...props
}: {
    className?: string,
    children: React.Node,
}) => (
    <a className={cx(style, className)} {...props}>
        {children}
    </a>
);
export default SupportTheGuardian;
