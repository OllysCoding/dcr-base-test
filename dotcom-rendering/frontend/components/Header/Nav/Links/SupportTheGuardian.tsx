import React from 'react';
import { cx, css } from 'react-emotion';

import { headline } from '@guardian/pasteup/fonts';
import { palette } from '@guardian/pasteup/palette';
const style = css`
    color: ${palette.nav.faded};
    font-family: ${headline};
    font-size: 14px;
    font-weight: 500;
    text-decoration: none;
    position: relative;
    width: 135px;
    display: block;
    text-align: center;
    padding-top: 6px;
    z-index: 0;
    float: left;
    :before {
        background-color: ${palette.neutral[7]};
        border-radius: 50%;
        top: -85px;
        left: 0;
        right: 0;
        padding-top: 100%;
        content: '';
        display: block;
        position: absolute;
        transition: background-color 250ms ease-out;
        z-index: -1;
    }
    :hover:before {
        background-color: ${palette.news.main};
        transform: scale(1.05);
    }
`;

const SupportTheGuardian: React.SFC<{
    className?: string;
    children: React.ReactChild;
    href: string;
}> = ({ className, children, href, ...props }) => (
    <a className={cx(style, className)} href={href} {...props}>
        {children}
    </a>
);
export default SupportTheGuardian;
