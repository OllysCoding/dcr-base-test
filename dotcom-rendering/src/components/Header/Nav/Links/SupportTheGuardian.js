// @flow

import styled from 'react-emotion';

import palette, { pillars } from 'pasteup/palette';

export default styled('a')(() => ({
    color: palette.neutral.header,
    fontFamily:
        '"Guardian Egyptian Web", "Guardian Text Egyptian Web", "Georgia", "serif"',
    fontWeight: 800,
    textDecoration: 'none',
    position: 'relative',
    width: 155,
    display: 'block',
    textAlign: 'center',
    paddingTop: 6,
    zIndex: 0,
    float: 'left',
    ':before': {
        backgroundColor: palette.neutral['1'],
        borderRadius: '50%',
        top: -100,
        left: 0,
        right: 0,
        paddingTop: '100%',
        content: '""',
        display: 'block',
        position: 'absolute',
        transition: 'background-color 250ms ease-out',
        zIndex: -1,
    },
    ':hover:before': {
        backgroundColor: pillars.news,
    },
}));
