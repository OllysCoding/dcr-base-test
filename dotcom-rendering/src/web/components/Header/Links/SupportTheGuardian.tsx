import React from 'react';
import { css } from 'emotion';
import { palette } from '@guardian/src-foundations';
import { headline } from '@guardian/src-foundations/typography';
import { from } from '@guardian/src-foundations/mq';
import { visuallyHidden } from '@guardian/src-foundations/accessibility';
import ProfileIcon from '@frontend/static/icons/profile.svg';
import { getCookie } from '@root/src/web/browser/cookie';
import { AsyncClientComponent } from '@root/src/web/components/lib/AsyncClientComponent';

const style = css`
    position: relative;
    display: block;
    float: left;
    margin-left: -20px;
    text-decoration: none;

    ${from.mobileLandscape} {
        margin-left: -10px;
    }

    :before {
        background-color: ${palette.neutral[7]};
        border-radius: 50%;
        bottom: -12px;
        left: 0;
        right: 0;
        padding-top: 100%;
        content: '';
        display: block;
        position: absolute;
        transition-timing-function: ease-out;
        transition-duration: 250ms;
        transition-property: background-color, transform;
    }

    :hover:before {
        background-color: ${palette.news.main};
        transform: scale(1.05);
    }
`;

const text = css`
    color: ${palette.neutral[97]};
    ${headline.tiny()};
    font-weight: 700;
    text-align: center;
    padding: 6px 20px 3px;
    position: relative;

    ${from.tablet} {
        ${headline.tiny()};
    }
`;

const mobileSignInContainer = css`
    display: block;
    padding: 10px;
    margin-left: -8px;
    float: left;

    ${from.mobileMedium} {
        padding-top: 12px;
    }

    ${from.tablet} {
        display: none;
    }
`;

const mobileSignInIcon = css`
    height: 18px;
    width: 18px;

    ${from.mobileMedium} {
        height: 23px;
        width: 23px;
    }

    fill: ${palette.neutral[46]};
`;

export const SupportTheGuardian: React.FC<{
    url: string;
}> = ({ url }) => (
    <AsyncClientComponent f={shouldShow}>
        {({ data }) => (
            <>
                {data && (
                    <>
                        <a className={style} href={url}>
                            <div className={text}>
                                Support The
                                <br /> Guardian
                            </div>
                        </a>
                        <div className={mobileSignInContainer}>
                            <a href="https://profile.theguardian.com/signin?INTCMP=DOTCOM_NEWHEADER_SIGNIN">
                                <ProfileIcon className={mobileSignInIcon} />
                                <span
                                    className={css`
                                        ${visuallyHidden};
                                    `}
                                >
                                    Sign in
                                </span>
                            </a>
                        </div>
                    </>
                )}
            </>
        )}
    </AsyncClientComponent>
);

const shouldShow: () => Promise<boolean> = () =>
    Promise.resolve(!(isRecentContributor() || isPayingMember()));

const isRecentContributor: () => boolean = () => {
    const value = getCookie('gu.contributions.contrib-timestamp');

    if (!value) {
        return false;
    }

    const now = new Date().getTime();
    const lastContribution = new Date(value).getTime();
    const diffDays = Math.ceil((now - lastContribution) / (1000 * 3600 * 24));

    return diffDays <= 180;
};

const isPayingMember: () => boolean = () => {
    return getCookie('gu_paying_member') === 'true';
};
