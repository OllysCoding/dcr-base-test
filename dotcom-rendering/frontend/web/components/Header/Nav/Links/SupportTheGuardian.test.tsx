import React from 'react';
import { render, wait } from 'react-testing-library';
import SupportTheGuardian from './SupportTheGuardian';
import {
    getCookie as getCookie_,
    addCookie as addCookie_,
} from '../../../../lib/cookie';

const getCookie: any = getCookie_;
const addCookie: any = addCookie_;

jest.mock('../../../../lib/cookie', () => ({
    getCookie: jest.fn(() => null),
    addCookie: jest.fn(() => null),
}));

describe('SupportTheGuardian', () => {
    const contributionsCookie = 'gu.contributions.contrib-timestamp';
    const payingMemberCookie = 'gu_paying_member';
    const url = 'https://www.theguardian.com';

    beforeEach(() => {
        addCookie.mockReset();
        getCookie.mockReset();
    });

    describe('snapshots', () => {
        it('should not render if recent contributor', async () => {
            const contributionDate = new Date();

            // set a contribution date of 1 week ago
            contributionDate.setDate(contributionDate.getDate() - 7);

            getCookie.mockReturnValue(contributionDate);

            const { container } = render(<SupportTheGuardian url={url} />);

            // expect nothing to be rendered
            await wait(() => expect(container.firstChild).toBeNull());

            expect(container.firstChild).toMatchSnapshot();
        });

        it('should not render correctly if paying member', async () => {
            const contributionDate = new Date();

            // set a contribution date of 1 year ago
            contributionDate.setDate(contributionDate.getDate() - 365);

            getCookie
                .mockReturnValueOnce(contributionDate)
                .mockReturnValueOnce('true');

            const { container } = render(<SupportTheGuardian url={url} />);

            // expect nothing to be rendered
            await wait(() => expect(container.firstChild).toBeNull());

            expect(container.firstChild).toMatchSnapshot();
        });

        it('should render if neither paying member or recent contributor', async () => {
            const contributionDate = new Date();

            // set a contribution date of 1 year ago
            contributionDate.setDate(contributionDate.getDate() - 365);

            getCookie
                .mockReturnValueOnce(contributionDate)
                .mockReturnValueOnce('false');

            const { container } = render(<SupportTheGuardian url={url} />);

            // expect something to be rendered
            await wait(() => expect(container.firstChild).not.toBeNull());

            expect(container.firstChild).toMatchSnapshot();
        });
    });

    it('should not render if recent contributor', async () => {
        const contributionDate = new Date();

        // set a contribution date of 1 week ago
        contributionDate.setDate(contributionDate.getDate() - 7);

        getCookie.mockReturnValue(contributionDate);

        const { container } = render(<SupportTheGuardian url={url} />);

        // expect nothing to be rendered
        await wait(() => expect(container.firstChild).toBeNull());

        expect(getCookie).toHaveBeenCalledTimes(1);
        expect(getCookie).toHaveBeenCalledWith(contributionsCookie);
    });

    it('should not render correctly if paying member', async () => {
        const contributionDate = new Date();

        // set a contribution date of 1 year ago
        contributionDate.setDate(contributionDate.getDate() - 365);

        getCookie
            .mockReturnValueOnce(contributionDate)
            .mockReturnValueOnce('true');

        const { container } = render(<SupportTheGuardian url={url} />);

        // expect nothing to be rendered
        await wait(() => expect(container.firstChild).toBeNull());

        expect(getCookie).toHaveBeenCalledTimes(2);
        expect(getCookie).toHaveBeenCalledWith(contributionsCookie);
        expect(getCookie).toHaveBeenCalledWith(payingMemberCookie);
    });

    it('should render if neither paying member or recent contributor', async () => {
        const contributionDate = new Date();

        // set a contribution date of 1 year ago
        contributionDate.setDate(contributionDate.getDate() - 365);

        getCookie
            .mockReturnValueOnce(contributionDate)
            .mockReturnValueOnce('false');

        const { container, getByText } = render(
            <SupportTheGuardian url={url} />,
        );

        // expect something to be rendered
        await wait(() => expect(container.firstChild).not.toBeNull());

        expect(getByText('Support The Guardian')).toBeInTheDocument();
        expect(getCookie).toHaveBeenCalledTimes(2);
        expect(getCookie).toHaveBeenCalledWith(contributionsCookie);
        expect(getCookie).toHaveBeenCalledWith(payingMemberCookie);
    });
});
