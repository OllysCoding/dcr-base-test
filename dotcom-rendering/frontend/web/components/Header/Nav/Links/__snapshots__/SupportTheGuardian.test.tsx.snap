// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupportTheGuardian snapshots should not render correctly if paying member 1`] = `null`;

exports[`SupportTheGuardian snapshots should not render if recent contributor 1`] = `null`;

exports[`SupportTheGuardian snapshots should render if neither paying member or recent contributor 1`] = `
.emotion-1 {
  position: relative;
  display: block;
  float: left;
  margin-left: -20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width:480px) {
  .emotion-1 {
    margin-left: -10px;
  }
}

.emotion-1:before {
  background-color: #121212;
  border-radius: 50%;
  bottom: -12px;
  left: 0;
  right: 0;
  padding-top: 100%;
  content: '';
  display: block;
  position: absolute;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 250ms;
  transition-duration: 250ms;
  -webkit-transition-property: background-color,-webkit-transform;
  -webkit-transition-property: background-color,transform;
  transition-property: background-color,transform;
}

.emotion-1:hover:before {
  background-color: #c70000;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.emotion-0 {
  color: #f6f6f6;
  font-family: GH Guardian Headline,Georgia,serif;
  font-size: 13px;
  font-weight: 700;
  text-align: center;
  line-height: 1.2;
  padding: 6px 20px 3px;
  position: relative;
}

@media (min-width:740px) {
  .emotion-0 {
    font-size: 14px;
  }
}

<a
  class="emotion-1"
  href="https://www.theguardian.com"
>
  <div
    class="emotion-0"
  >
    Support The
    <br />
     Guardian
  </div>
</a>
`;
